docs/usage/mnist.rst

```diff
@@ -0,0 +1,92 @@
+      | gzip -d - > t10k-images-idx3-ubyte
+    $ curl -sSL 'http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz' \
+      | gzip -d - > t10k-labels-idx1-ubyte
+
+.. code-block:: console
+
+		$ dffml train \
+				-model tfdnnc \
+				-model-batchsize 1000 \
+				-model-hidden 30 50 25 \
+				-model-clstype int \
+				-model-classification label \
+				-model-classifications $(seq 0 9) \
+				-model-features def:image:int:$((28 * 28)) \
+				-sources images=idx3 label=idx1 \
+				-source-images-filename train-images-idx3-ubyte \
+				-source-images-feature image \
+				-source-label-filename train-labels-idx1-ubyte \
+				-source-label-feature label \
+				-log debug
+		... log output ...
+		$ dffml accuracy \
+				-model tfdnnc \
+				-model-batchsize 1000 \
+				-model-hidden 30 50 25 \
+				-model-clstype int \
+				-model-classification label \
+				-model-classifications $(seq 0 9) \
+				-model-features def:image:int:$((28 * 28)) \
+				-sources images=idx3 label=idx1 \
+				-source-images-filename t10k-images-idx3-ubyte \
+				-source-images-feature image \
+				-source-label-filename t10k-labels-idx1-ubyte \
+				-source-label-feature label \
+				-log debug
+    ... log output ...
+    0.6417230000123672
+
+.. TODO Everything that follows this TODO message is what still needs to be done
+
+To get good accuracy on this problem the image matrices need to have each
+element scaled from 0 to 1 (normalize them). I think what probably makes sense
+here it to create a new source within the main dffml codebase (so not in a
+plugin) which has the real source as a property of its config.
+
+.. code-block:: console
+
+		$ dffml train \
+				-model tfdnnc \
+				-model-batchsize 1000 \
+				-model-hidden 30 50 25 \
+				-model-clstype int \
+				-model-classification label \
+				-model-classifications $(seq 0 9) \
+        `# BEGIN new stuff` \
+				-model-features def:image:float:$((28 * 28)) \
+				-sources images=preprocess label=idx1 \
+				-source-images-source idx3 \
+				-source-images-source-filename train-images-idx3-ubyte \
+				-source-images-source-feature image \
+				-source-images-transform image:array_divide \
+        `# END new stuff` \
+				-source-label-filename train-labels-idx1-ubyte \
+				-source-label-feature label \
+				-log debug
+
+The issue I see us hitting first is that we need to find a good way to tell the
+``preprocess`` source what operations should be given to what features. And we
+need a way to specify what input gets what value. There will probably be some
+editing of the ``config_get`` code...
+
+This is just for quick and dirty pre-processing. Later we'll want to add a
+``dfpreprocess`` to use a dataflow to do pre-processing. For this example that
+would be too much overhead. Creating something that normalizes the dataset just
+by adding a few more CLI arguments will be very powerfull through it's
+simplicity.
+we need to figure out how to put a dataflow in front of
```

Comment:
I think what we'll need to do is modify the config code in dffml/base py to be able to understand a typing.Dict value where the key is a str type, and then load the subsequent values based on their type. I'll try to explain more later.